ci/test_wheel_deps_wheels.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+# Copyright (c) 2023-2024, NVIDIA CORPORATION
+
+set -euo pipefail
+
+rapids-logger "Install testing dependencies"
+# TODO: Replace with rapids-dependency-file-generator
+python -m pip install \
+    psutil \
+    cffi \
+    cuda-python \
+    nvidia-curand-cu12 \
+    nvidia-cuda-nvcc-cu12 \
+    nvidia-cuda-nvrtc-cu12 \
+    pynvjitlink-cu12 \
+    pytest
+
+
+rapids-logger "Build tests"
+PY_SCRIPT="
+import numba_cuda
+root = numba_cuda.__file__.rstrip('__init__.py')
+test_dir = root + \"numba/cuda/tests/test_binary_generation/\"
+print(test_dir)
+"
+
+NUMBA_CUDA_TEST_BIN_DIR=$(python -c "$PY_SCRIPT")
+pushd $NUMBA_CUDA_TEST_BIN_DIR
+make
+popd
+
+rapids-logger "Install wheel"
+package=$(realpath wheel/numba_cuda*.whl)
+echo "Package path: $package"
+python -m pip install $package
+
+rapids-logger "Check GPU usage"
+nvidia-smi
+
+RAPIDS_TESTS_DIR=${RAPIDS_TESTS_DIR:-"${PWD}/test-results"}/
+mkdir -p "${RAPIDS_TESTS_DIR}"
+pushd "${RAPIDS_TESTS_DIR}"
+
+rapids-logger "Show Numba system info"
+python -m numba --sysinfo
+
+# remove cuda-nvvm-12-5 leaving libnvvm.so from nvidia-cuda-nvcc-cu12 only
+apt-get update
+apt remove --purge `dpkg --get-selections | grep cuda-nvvm | awk '{print $1}'` -y
+apt remove --purge `dpkg --get-selections | grep cuda-nvrtc | awk '{print $1}'` -y
```

Comment:
Do we plan to make CUDA wheels a hard dependency for numba-cuda? (I hope so.)

Also, to help with testing this, we can create some images in `rapidsai/ci-imgs` that are like `citestwheel` but using `nvidia/cuda` `base` flavor, which will make it possible to test cases where CUDA wheels are the only source of CUDA libraries. Please feel free to ask for help with this if it would help you.

Reply:
If it isn't too much overhead then this would be cleaner I agree, that said, I think it's not the worst outcome to have something that just works here for now. When `cuda-python` becomes a hard dependency, it will handle all the pathfinding, so part of me thinks the long term investment in CI infra for testing these supported library configurations should also just be there.

Reply:
Okay, let's defer on this until later.

Reply:
`cuda-python` itself doesn't take a hard dependency on the CUDA wheels: https://github.com/NVIDIA/cuda-python/blob/main/cuda_bindings/pyproject.toml#L38

Regardless, this should punt the problem to `cuda-python` and we should align on how we want to handle it there.